src/scripts/create-user.ts

```diff
@@ -1,45 +1,46 @@
-import { CreateUserRequestOptions } from '@okta/okta-sdk-nodejs';
 import { Argv } from 'yargs';
 import { RootCommand } from '..';
 import { generate } from 'generate-password';
 
-import { oktaUserAsUser, User, getUser } from './services/user-service';
-import { oktaManageClient, OktaConfiguration } from './services/client-service';
+import { User, UserService, OktaUserService } from './services/user-service';
+import { oktaManageClient } from './services/client-service';
 
-const createUser = async (
-  oktaConfiguration: OktaConfiguration,
+import * as TE from 'fp-ts/lib/TaskEither';
+import * as E from 'fp-ts/lib/Either';
+import { pipe } from 'fp-ts/lib/function';
+import * as Console from 'fp-ts/lib/Console';
+
+const createUser = (
+  service: UserService,
   password: string,
   email: string,
   firstName: string,
   lastName: string
-): Promise<User> => {
-  const client = oktaManageClient(oktaConfiguration);
-
-  const maybeOktaUser = await getUser(email, client);
-  const newUser: CreateUserRequestOptions = {
-    profile: {
-      firstName: firstName,
-      lastName: lastName,
-      email: email,
-      login: email,
-    },
-    credentials: {
-      password: { value: password },
-    },
-  };
-
-  // eslint-disable-next-line functional/functional-parameters
-  const throwOnExisting = () => {
-    // eslint-disable-next-line functional/no-throw-statement
-    throw new Error(
-      `User [${email}] already exists. Can not create a new user.`
-    );
-  };
-
-  return maybeOktaUser === undefined
-    ? oktaUserAsUser(await client.createUser(newUser))
-    : throwOnExisting();
-};
+): TE.TaskEither<string, User> =>
+  pipe(
+    email,
+    service.getUser,
+    TE.chain((maybeUser) =>
+      pipe(
+        service.validateUserExists(maybeUser, email),
+        // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
+        (validated) =>
+          E.isLeft(validated)
+            ? service.createUser(email, firstName, lastName, password)
+            : TE.chain((user: User) =>
```

Comment:
You can remove nearly all of this when you stop using `validateUserExists`.

Reply:
i.e., as it was before:

```
    TE.chain(
      flow(
        O.fold(
          // eslint-disable-next-line functional/functional-parameters
          () => service.createUser(email, firstName, lastName, password),
          // eslint-disable-next-line @typescript-eslint/prefer-readonly-parameter-types
          (user) =>
            TE.left(
              `User [${email}] already exists with id [${user.id}]. Can not create a new user.`
            )
        )
      )
    )
```

Reply:
replaced with fold